### PR TITLE
Dev kmr

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Review by rafaelfolco — DISMISSED** [review]

**Comment by pradiptapks** [comment]
We may need a further review as without `trex-config` returned below exception.

```
trex-config in param file.
# cat run/iterations/iteration-1-fail/sample-1-fail-1/client/1/trex-server-stderrout.txt
Starting  TRex v2.87 please wait  ... 
Using configuration file /tmp/tmp.V32VOyrUfl/iteration-1/sample-1/trex_cfg.yaml 
 port limit     :  not configured 
 port_bandwidth_gb    :  10 
 if_mask        : None 
 is low-end : 0 
 stack type :  
 thread_per_dual_if      : 1 
 if        : 
 m_zmq_rpc_port    :  4501 
 memory per 2x10G ports  
 MBUF_64                                   : 16380 
 MBUF_128                                  : 8190 
 MBUF_256                                  : 8190 
 MBUF_512                                  : 8190 
 MBUF_1024                                 : 8190 
 MBUF_2048                                 : 4095 
 MBUF_4096                                 : 128 
 MBUF_9K                                   : 512 
 TRAFFIC_MBUF_64                           : 65520 
 TRAFFIC_MBUF_128                          : 32760 
 TRAFFIC_MBUF_256                          : 8190 
 TRAFFIC_MBUF_512                          : 8190 
 TRAFFIC_MBUF_1024                         : 8190 
 TRAFFIC_MBUF_2048                         : 32760 
 TRAFFIC_MBUF_4096                         : 128 
 TRAFFIC_MBUF_9K                           : 512 
 MBUF_DP_FLOWS                             : 524288 
 MBUF_GLOBAL_FLOWS                         : 5120 
 no platform info 
CTimerWheelYamlInfo does not exist  
ERROR need at least 2 ports 
```

**Comment by k-rister** [comment]
That issue should be resolved in the latest version.  However, I'm working on a more elegant solution so for the time being I've marked this PR as a draft.

File: trafficgen-infra [hunk 1]
```diff
@@ -20,11 +20,18 @@ rate_unit="%"
...
+
+longopts="active-devices:,devices:,cpus:,mem-limit:,trex-software-mode:"
 opts=$(getopt -q -o "" --longoptions "$longopts" -- "$@");
 eval set -- "$opts";
 while true; do
     case "$1" in
+	--trex-software-mode)
+	    shift
+	    trex_software_mode=${1}
+	    shift
```
Comment (rafaelfolco) [inline trafficgen-infra:39]: is this right?

**Review by rafaelfolco — APPROVED** [review]
lgtm

File: trafficgen-infra (hunk 1 above)
Comment (k-rister) [reply trafficgen-infra:39]: Yes.  If you look at the existing parameters they all follow a similar form.  What looks weird/wrong about it?

**Pull request merged**